Wed Aug 2 2006 Stuart Hughes <stuarth@freescale.com>

This patch is a bit of a hack that allows the specification in /etc/inittab
of '0' as the baud rate.  If this is given, then the current setting is
used.  This is so I can have a generic entry in /etc/inittab that will 
work for all boards:

::respawn:/sbin/getty console 0 vt100


diff --exclude CVS -uNr busybox-1.1.3/loginutils/getty.c busybox-1.1.3.modified/loginutils/getty.c
--- busybox-1.1.3/loginutils/getty.c	2006-03-22 21:16:24.000000000 +0000
+++ busybox-1.1.3.modified/loginutils/getty.c	2006-08-02 15:18:50.000000000 +0100
@@ -432,11 +432,22 @@
 static void parse_speeds(struct options *op, char *arg)
 {
 	char *cp;
+	speed_t speed;
 
 	debug("entered parse_speeds\n");
 	for (cp = strtok(arg, ","); cp != 0; cp = strtok((char *) 0, ",")) {
-		if ((op->speeds[op->numspeed++] = bcode(cp)) <= 0)
+		if( (speed = bcode(cp)) >= 0) {
+			if(speed == 0) {
+				struct termio tio;
+				if (ioctl(0, TCGETA, &tio) < 0) {
+					error("bcode ioctl(0, TCGETA, &tio) failed\n");
+				}
+				speed = cfgetospeed(&tio);
+			}
+			op->speeds[op->numspeed++] = speed;
+		} else {
 			error("bad speed: %s", cp);
+		}
 		if (op->numspeed > MAX_SPEED)
 			error("too many alternate speeds");
 	}
